### DISCUSSION TITLE
I don't have enough time until_after_source

### DISCUSSION THREAD

**Discussion by leSemaleon** [NanoComp/meep#2677] [Q&A]
I want to calculate the transmission spectrum. I use the stop_when_fields_decayed function. With my simulation size, the field from the gaussiansource does not have time to reach the point at which I check the byDecay before the source stops emitting. This causes the simulation to end prematurely. It is possible to add a constant term to until_after_source?

**Comment by smartalecH** [comment]
> It is possible to add a constant term to until_after_source

yes… and certainly trying that is much easier than writing a new question about it… so maybe I don’t understand your actual question?

**Reply by leSemaleon** [reply]
Ok, I'll try to explain again. I know that if we write number after "until_after_source" this will be an additional addition of time. But i want to use condition stop_when_fields_decayed after source and some more time

**Reply by leSemaleon** [reply]
Do you know, why this happens?
